commercetools/resource_custom_object.go

```diff
@@ -68,13 +69,43 @@ func resourceCustomObjectCreate(d *schema.ResourceData, m interface{}) error {
 }
 
 func resourceCustomObjectRead(d *schema.ResourceData, m interface{}) error {
+	container := d.Get("container").(string)
+	key := d.Get("key").(string)
+	log.Printf("[DEBUG] Reading custom object from commercetools with following values\n Container: %s \n Key: %s", container, key)
+	client := getClient(m)
+
+	customObject, err := client.CustomObjectGetWithContainerAndKey(context.Background(), container, key)
+	if err != nil {
+		if ctErr, ok := err.(commercetools.ErrorResponse); ok {
+			if ctErr.StatusCode == 404 {
+				d.SetId("")
+				return nil
+			}
+		}
+		return err
+	}
+
+	if customObject == nil {
+		log.Print("[DEBUG] No custom object found")
+		d.SetId("")
+	} else {
+		log.Print("[DEBUG] Found following custom object:")
+		log.Print(stringFormatObject(customObject))
+		d.Set("container", customObject.Container)
+		d.Set("key", customObject.Key)
+		d.Set("value", customObject.Value)
+		d.Set("version", customObject.Version)
+	}
 	return nil
 }
 
 func resourceCustomObjectUpdate(d *schema.ResourceData, m interface{}) error {
 	client := getClient(m)
 	value := _decodeCustomObjectValue(d.Get("value").(string))
 	ctx := context.Background()
+	originalKey, _ := d.GetChange("key")
+	originalContainer, _ := d.GetChange("container")
+	originalVersion, _ := d.GetChange("version")
```

Comment:
If you also grab the new values from the `d.getChange()` call you can save yourself 3 d.Get() calls on lines 142 to 145
```suggestion
	originalKey, newKey := d.GetChange("key")
	originalContainer, newContainer := d.GetChange("container")
	originalVersion, newVersion := d.GetChange("version")
```